Integration tests for plugins fail on Windows platforms

Integration tests for plugins are failing on Windows platforms because:
- it tries to print message using a wrong format (see #16367)
- when installing the plugin using a file, it mixes up the file path (ex: `file://C:\whatever`) and thinks it refers to a plugin to download on Maven Central (ex: `org.elasticsearch:mapper-attachments:3.0.0`) because of the number of `:` in the name

```
....
bin\plugin install file:/Y:/jenkins/workspace/es_core_master_window-2012/plugins/analysis-icu/build/cluster/integTest%20node0/plugins%20tmp/analysis-icu-3.0.0-SNAPSHOT.zip
Successfully started process 'command 'cmd''
Plugins directory [Y:\jenkins\workspace\es_core_master_window-2012\plugins\analysis-icu\build\cluster\integTest node0\elasticsearch-3.0.0-SNAPSHOT\plugins] does not exist. Creating...
Exception in thread "main" java.util.IllegalFormatWidthException: 20
	at java.util.Formatter$FormatSpecifier.checkText(Formatter.java:3044)
	at java.util.Formatter$FormatSpecifier.<init>(Formatter.java:2733)
	at java.util.Formatter.parse(Formatter.java:2560)
	at java.util.Formatter.format(Formatter.java:2501)
	at java.util.Formatter.format(Formatter.java:2455)
	at java.lang.String.format(String.java:2981)
	at org.elasticsearch.common.cli.Terminal$SystemTerminal.doPrint(Terminal.java:161)
	at org.elasticsearch.common.cli.Terminal.print(Terminal.java:110)
	at org.elasticsearch.common.cli.Terminal.println(Terminal.java:105)
	at org.elasticsearch.common.cli.Terminal.println(Terminal.java:93)
	at org.elasticsearch.plugins.InstallPluginCommand.download(InstallPluginCommand.java:168)
	at org.elasticsearch.plugins.InstallPluginCommand.execute(InstallPluginCommand.java:140)
	at org.elasticsearch.common.cli.CliTool.execute(CliTool.java:145)
	at org.elasticsearch.plugins.PluginCli.main(PluginCli.java:74)
:plugins:analysis-icu:integTest#installAnalysisIcuPlugin FAILED
:plugins:analysis-icu:integTest#installAnalysisIcuPlugin (Thread[main,5,main]) completed. Took 1.304 secs.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':plugins:analysis-icu:integTest#installAnalysisIcuPlugin'.
> Process 'command 'cmd'' finished with non-zero exit value 1
```

--- a/core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java
+++ b/core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java
@@ -160,9 +160,9 @@ class InstallPluginCommand extends CliTool.Command {
             return downloadZipAndChecksum(url, tmpDir);
         }
 
-        // now try as maven coordinates, a valid URL would only have a single colon
+        // now try as maven coordinates, a valid URL would only have a colon and slash
         String[] coordinates = pluginId.split(":");
-        if (coordinates.length == 3) {
+        if (coordinates.length == 3 && pluginId.contains("/") == false) {
             String mavenUrl = String.format(Locale.ROOT, "https://repo1.maven.org/maven2/%1$s/%2$s/%3$s/%2$s-%3$s.zip",
                 coordinates[0].replace(".", "/") /* groupId */, coordinates[1] /* artifactId */, coordinates[2] /* version */);
             terminal.println("-> Downloading " + pluginId + " from maven central");

--- a/qa/evil-tests/src/test/java/org/elasticsearch/plugins/InstallPluginCommandTests.java
+++ b/qa/evil-tests/src/test/java/org/elasticsearch/plugins/InstallPluginCommandTests.java
@@ -20,6 +20,7 @@
 package org.elasticsearch.plugins;
 
 import java.io.IOException;
+import java.net.MalformedURLException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
 import java.nio.file.DirectoryStream;
@@ -203,6 +204,14 @@ public class InstallPluginCommandTests extends ESTestCase {
         Files.copy(new URL(pluginZip).openStream(), pluginZipWithSpaces, StandardCopyOption.REPLACE_EXISTING);
         installPlugin(pluginZipWithSpaces.toUri().toURL().toString(), env);
         assertPlugin("fake", pluginDir, env);
+    }
+
+    public void testMalformedUrlNotMaven() throws Exception {
+        // has two colons, so it appears similar to maven coordinates
+        MalformedURLException e = expectThrows(MalformedURLException.class, () -> {
+            installPlugin("://host:1234", createEnv());
+        });
+        assertTrue(e.getMessage(), e.getMessage().contains("no protocol"));
     }
 
     public void testPluginsDirMissing() throws Exception {

